feat(ui): add comprehensive dark mode support to dashboard

CryptoCanuck opened the pull request:
## Relevant issues

Closes #10177

## Linear ticket

N/A

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [x] I have added meaningful tests
- [x] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] My PR's scope is as isolated as possible; it only solves 1 specific problem
- [x] I have requested a Greptile review by commenting `@greptileai` and received a **Confidence Score of at least 4/5** before requesting a maintainer review

## Type

🆕 New Feature

## Changes

This enables dark mode for the admin dashboard. The navbar already shipped a dark-mode toggle that was scaffolded but gated off behind `{false && ...}`, with a note to keep it disabled until the dashboard supported dark styles end to end; this wires that toggle up and makes it work.

A shared `useDarkMode` hook owns the boolean. It persists the choice to `localStorage`, falls back to the OS `prefers-color-scheme` on first load, and toggles the `dark` class on the document element. The App Router dashboard layout, the legacy root page, and the public model hub all consume the same hook, so the toggle behaves consistently no matter which surface renders the navbar. Previously the legacy page kept its own local dark-mode state that did not persist or set the `dark` class, so Tremor components stayed light; routing it through the hook fixes that.

Ant Design is themed centrally through `getAntdTheme` on a `ConfigProvider` (dark algorithm plus component tokens for Menu, Table, Input, Modal, and friends). Tremor has no `ConfigProvider`, so its dark styling lives in `globals.css` keyed off the `.dark` class. An inline pre-hydration script in the root layout sets the `dark` class before first paint to avoid a flash of light mode.

The work targets `litellm_oss_branch` per the contribution flow for external PRs. The branch carries only UI changes; no `uv.lock`, dependency, or Python files are touched.

## Screenshots / Proof of Fix

To verify locally, run the dashboard and flip the Sun/Moon toggle in the navbar (top right). The selected mode persists across reloads, a fresh session follows the OS setting, and there is no flash of light mode on load. Worth checking the models table, logs, and the playground.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/BerriAI/litellm?pullRequest=18293) <br/>All committers have signed the CLA.

CryptoCanuck committed:
feat(ui): enable dashboard dark mode

Enable the dark-mode toggle that was scaffolded but gated off in the navbar, and
make it work end to end. A shared useDarkMode hook owns the boolean, persists it
to localStorage, follows the OS prefers-color-scheme on first load, and toggles
the `dark` class on the document element; the dashboard layout, the legacy root
page, and the public model hub all consume it so the toggle stays consistent
across surfaces. AntD is themed through getAntdTheme on a ConfigProvider; Tremor,
which has no ConfigProvider, is themed via `.dark` rules in globals.css. A
pre-hydration script in the root layout sets the class before paint to avoid a
flash of light mode.

Closes #10177

CryptoCanuck commented:
@greptileai

CryptoCanuck commented on ui/litellm-dashboard/src/app/page.tsx:
```diff
@@ -450,9 +447,7 @@ function CreateKeyPageContent() {
-          algorithm: isDarkMode ? theme.darkAlgorithm : theme.defaultAlgorithm,
-        }}>
+      <ConfigProvider theme={getAntdTheme(isDarkMode)}>
           <ThemeProvider accessToken={accessToken}>
```
Good catch. Fixed by making ThemeProvider the single owner of dark-mode state: it now wraps CreateKeyPageContent at the export, and the component reads isDarkMode/toggleDarkMode via useTheme() for both the ConfigProvider and the Navbar. The duplicate useDarkMode() instance is gone, so there's one source of truth.

CryptoCanuck commented on ui/litellm-dashboard/src/app/globals.css:
```diff
@@ -28,12 +57,1775 @@ body {
+
+.dark .bg-green-50 {
+  background-color: rgba(34, 197, 94, 0.1) !important;
+}
+
+.dark .bg-green-100 {
+  background-color: rgba(34, 197, 94, 0.2) !important;
+}
+
+.dark .hover\:bg-green-50:hover {
```
Agreed. Scoped these to the neutral gray shades (border-b/border-t combined with border-gray-100/200/300) so semantic colored directional borders like border-b border-blue-500 keep their color in dark mode.

CryptoCanuck committed:
fix(ui): refine dark mode palette and theme remaining surfaces

- Deepen the dark palette and quiet borders (base #0e0e0e, surfaces
  #151515, default border #1e1e1e) across globals.css, antdTheme, navbar
  and the dashboard layout.
- Add a dark default border color in @layer base so colorless `border`
  utilities stop rendering as light gray boxes.
- Scope the Tremor Badge and Switch overrides to the component root so the
  inner text span no longer gets its own grey background.
- Add dark overrides for bg-gray-200/300 chips and disabled buttons.
- Make the Workflow Runs page (fully inline-styled) theme-aware via a
  palette driven by the active theme.

CryptoCanuck commented:
@greptileai

CryptoCanuck committed:
fix(ui): resolve dark mode dual-instance and theming in public model hub

Address Greptile review feedback on the public model hub:
- Consolidate to a single dark-mode source: PublicModelHub now wraps a
  PublicModelHubContent child in ThemeProvider and the content reads state
  via useTheme(), removing the second standalone useDarkMode() instance
  that ran alongside the provider.
- Wrap the page in ConfigProvider(getAntdTheme(isDarkMode)) so the antd
  Modals, Selects, Tabs and Tags theme correctly, matching the dashboard
  and legacy page.tsx routes.
- Add an explicit dark variant on the outer container background.

CryptoCanuck commented:
Thanks for the review. Addressed both points in 5737b5e.

The duplicate `useDarkMode()` is gone — `PublicModelHub` now wraps `PublicModelHubContent` in `ThemeProvider` and the content reads `isDarkMode`/`toggleDarkMode` via `useTheme()`, so there's a single source of truth like the dashboard and `page.tsx` routes. I also wrapped the page in `ConfigProvider(getAntdTheme(isDarkMode))` so the antd modals/selects/tabs theme correctly in dark mode, which it was missing before.

On the outer `bg-white` — that one's actually already handled by globals.css, which has a global `.dark .bg-white` override (plus `.dark [class*="tremor-Card"]` for the cards), so the hub renders dark rather than white. Confirmed on the standalone `/ui/model_hub_table` page. I added an explicit `dark:bg-[#0e0e0e]` on the container anyway so the intent is clear in the markup.

CryptoCanuck commented:
<img width="1892" height="886" alt="litellm-dark-2" src="https://github.com/user-attachments/assets/676ad999-4788-46f2-bd6b-fe5addaf5ea6" />
<img width="1890" height="883" alt="litellm-dark-3" src="https://github.com/user-attachments/assets/9d891595-ee98-48cc-a3cf-bf2c8500223f" />
<img width="1888" height="884" alt="litellm-dark-4" src="https://github.com/user-attachments/assets/3ed2da45-896d-4996-9dbb-396b589a14a2" />
<img width="1902" height="883" alt="litellm-dark-1" src="https://github.com/user-attachments/assets/f03aff61-5ed9-4f4c-b3c5-3a05a0884263" />

CryptoCanuck commented:
Closing to push the adoption of the updated work on #35615
